cmd/snap-confine/snap-confine.c

```diff
@@ -347,11 +333,104 @@ int main(int argc, char **argv) {
     debug("ruid: %d, euid: %d, suid: %d", real_uid, effective_uid, saved_uid);
     debug("rgid: %d, egid: %d, sgid: %d", real_gid, effective_gid, saved_gid);
 
-    // snap-confine needs to run as root for cgroup/udev/mount/apparmor/etc setup.
-    if (effective_uid != 0) {
-        die("need to run as root or suid");
+    struct sc_apparmor apparmor;
```

Comment:
Why are we doing this earlier now?

Reply:
If I understand Alberto's original idea correctly, it's more of a consistency check and useful to ensure before we start parsing arguments in sc_init_invocation()